src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -367,4 +369,88 @@ public boolean contains(String key) {
+                    && tokenCacheItem.getFamilyClientId().equalsIgnoreCase("1")){
+                tokenItem = tokenCacheItem;
+                //return the cacheItem which contains the FRT for this userId
+                //there should be at most only one FRT record for each userId
+                return tokenItem;
+            }
+        }
+        //If the cache does not contain the FRT for this user return null
+        return null;
+    }
+    
+    public String serialize(String uniqueUserId) throws AuthenticationException{
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if((new BrokerProxy(mContext)).canSwitchToBroker()){
+            throw new AuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
+        } else {
+            TokenCacheItem tokenItem = getFamilyRefreshTokenItemForUser(uniqueUserId);
+            
+            if(tokenItem!=null) {
+                BlobContainer blobContainer = new BlobContainer(tokenItem);
+                Gson gson = new Gson();  
+                
+                if (StringExtensions.IsNullOrBlank(blobContainer.getFamilyRefreshToken())) {
+                    Logger.i(TAG, "serialization of FRT cache item: family refresh token is null", "");
+                    //throw new AuthenticationException(ADALError.FAIL_TO_EXPORT,"serialization of FRT cache item: family refresh token is null");
+                    return null;
+                }
+                
+                Logger.i(TAG, "family refresh token cache item:" + blobContainer.toString(), "");
+                String json = gson.toJson(blobContainer);
+                //should json be encrypted?
+                return json;
+                
+            } else {
+                Logger.i(TAG, "no FRT is found for this user" + uniqueUserId, "");
+                return null;
+            }
+        }      
+    }
+    
+    public void deserialize(String serializedBlob) throws AuthenticationException{
+        if (StringExtensions.IsNullOrBlank(serializedBlob)) {
+            throw new IllegalArgumentException("serializedBlob");
+        }
+            
+        if((new BrokerProxy(mContext)).canSwitchToBroker()) {
+            throw new AuthenticationException(ADALError.FAIL_TO_IMPORT,"Failed to import the serialized blob because broker is enabled.");
+        } else {
+            Gson gson = new Gson();  
+            BlobContainer blobContainer;
+            
+            try {
+                blobContainer = gson.fromJson(serializedBlob, BlobContainer.class);
+            } catch (JsonSyntaxException exp) {
+                throw new AuthenticationException(ADALError.FAIL_TO_IMPORT,
+                        exp.getMessage());
+            }
+            //@Heidi
+            //Q. how to compare the SerialVersionUID?
```

Comment:
Why do you need this check at all? If json failed to parse it means types are incompartible and failed to deserialize
If it didn't fail on parsing, no need to track this serialVersion